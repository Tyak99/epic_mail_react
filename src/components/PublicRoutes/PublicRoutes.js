import React from 'react';
import PropTypes from 'prop-types';
import { Route, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';

export class PublicRoutes extends React.Component {
  constructor(props) {
    super(props);
    this.state = { };
  }

  render() {
    const { component: Component, isAuthenticated, ...rest } = this.props;

    return (
      <Route
        {...rest}
        render={props => (!isAuthenticated ? <Component {...props} /> : <Redirect to="/inbox" />)}
      />
    );
  }
}

PublicRoutes.propTypes = {
  component: PropTypes.element,
  isAuthenticated: PropTypes.bool.isRequired,
};

const mapStateToProps = state => ({
  isAuthenticated: state.auth.isAuthenticated,
});
export default connect(mapStateToProps)(PublicRoutes);
