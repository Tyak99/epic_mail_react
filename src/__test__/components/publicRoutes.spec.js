import React from 'react';
import { shallow } from 'enzyme';
import { PublicRoutes } from '../../components/PublicRoutes/PublicRoutes';

const props = {
  isAuthenticated: true,
  path: '/',
};

describe('component: PrivateRoute', () => {
  it('should render correctly', (done) => {
    const wrapper = shallow(<PublicRoutes {...props} />);
    expect(wrapper).toMatchSnapshot();
    done();
  });
});
