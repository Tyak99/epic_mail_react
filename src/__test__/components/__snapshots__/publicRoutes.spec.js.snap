// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: PrivateRoute should render correctly 1`] = `ShallowWrapper {}`;
